libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -378,7 +380,7 @@ void ReferencesResolver::endVisit(VariableDeclaration const& _variable)
 	else if (_variable.isStateVariable())
 	{
 		solAssert(varLoc == Location::Unspecified, "");
-		typeLoc = _variable.isConstant() ? DataLocation::Memory : DataLocation::Storage;
+		typeLoc = (_variable.isConstant() || _variable.immutable()) ? DataLocation::Memory : DataLocation::Storage;
```

Comment:
Why does `_variable.immutable()` makes the type location `Memory`?

Reply:
If we at some point implement immutable arrays, then they will either be in memory, or we add the data location "code", but currently, this is the closest match.